### DISCUSSION TITLE
Permissions of the lease file

### DISCUSSION THREAD

**Discussion by semihalf-ostruszka-andrzej** [rsmarples/dhcpcd#132] [Ideas]
Hi

I'd like to ask about possibility to change lease file permissions from 0600 to 0660 so that group members have write permissions.
The context of this question is that in ChromeOS dhcpcd is being spawned by the 'shill' (ChromeOS "connection manager") and we spawn dhcpcd with 'dhcp'/'dhcp' uid/gid but would like the 'shill' to be able to remove lease file.  So currently we will have 'shill' as a member of 'dhcp' group and we have tiny patch changing lease file creation mask and we are curious if that change could be upstream'ed.

With regards
Andrzej

**Comment by ido** [comment]
While I don't object to having a `leasefilemode` configuration parameter, can't you already achieve the same effect using dhcpcd hooks? Have you considered doing the `chmod` (or better yet, `setfacl`) of the lease file in a hook script?

**Comment by rsmarples** [comment]
What version are you using? Current dhcpcd writes using 0640 permissions so group can at least read.

I don't see why that should be changed. If you want dhcpcd to remove the lease file then terminate it with `-k` or `SIGALRM`. Or remove the `persistent` directive from `/etc/dhcpcd.conf`.

Is there a reason why these methods are unacceptable?

**Comment by semihalf-ostruszka-andrzej** [comment]
Thanks Ido for the suggestion.  That would be an option but ... in ChromeOS dhcpcd is used with this functionality disabled (custom patch).

**Comment by semihalf-ostruszka-andrzej** [comment]
Hi Roy

In ChromeOS we use dhcpcd in a bit unconventional way.  Since large portion of the platform part of dhcpcd is being actually handled outside of the dhcpcd (routing, link management etc. is handled in ChromeOS connection manager called "shill") the dhcpcd is only used for DHCP lease acquisition - the result is then passed over DBus to "shill".  So the dhcpcd is not a single service being run for multiple interfaces but its lifetime is bound to the lifetime of the interface/connection (dhcpcd is being spawned and killed by the "shill").  So at the moment we decide to change MAC address there is no dhcpcd to be killed/signaled.

However, having said that, it turns out that it is actually not really necessary to have write permissions to remove the lease file - only write permissions to the directory is enough (my mistake).  So I guess we can stop here (although I like Ido's idea about config param `leasefilemode`).

Regards
Andrzej

**Reply by ido** [reply]
I'm resolving thiis since you don't need to be able to set leasefile modes after all, but please reopen and @ me if you end up needing `leasefilemode` after all.
